src/Illuminate/Foundation/Exceptions/Handler.php

```diff
@@ -1072,4 +1079,36 @@ protected function newLogger()
     {
         return $this->container->make(LoggerInterface::class);
     }
+
+    /**
+     * Set a callback to build the resulting exception for a ModelNotFoundException.
+     *
+     * @param  string|null|\Closure(\Illuminate\Database\Eloquent\ModelNotFoundException):  \Throwable   $callback
+     * @return $this
+     */
+    public function setModelNotFoundCallback($callback)
+    {
+        if (is_string($callback)) {
+            $callback = fn ($e) => new NotFoundHttpException($callback, $e);
+        }
+
+        $this->modelNotFoundResponseCallback = $callback;
+
+        return $this;
+    }
+
+    /**
+     * Build a Throwable from a ModelNotFoundException.
+     *
+     * @param  ModelNotFoundException  $e
+     * @return \Throwable
+     */
+    protected function mapModelNotFoundException($e)
+    {
+        if (isset($this->modelNotFoundResponseCallback)) {
+            return call_user_func($this->modelNotFoundResponseCallback, $e);
+        }
+
+        return new NotFoundHttpException($e->getMessage(), $e);
+    }
```

Comment:
> ```php
>         if (is_string($callback)) {
>            $callback = fn ($e) => new NotFoundHttpException($callback, $e);
>         }
> ```
`string` in combination with `$callback` is usually interpreted as a `callable`, so this can be confusing.

Furthermore, I don't think, passing an arbitrary string is a good improvement. Though, you have a point about customizability. But it would be neater if one could pass something like this:
```suggestion
    /**
     * Set a callback to build the resulting exception for a ModelNotFoundException.
     *
     * @param  class-string|null|\Closure(\Illuminate\Database\Eloquent\ModelNotFoundException):  \Throwable   $callback
     * @return $this
     */
    public function setModelNotFoundCallback($callback = NotFoundHttpException::class)
    {
        if (class_exists($callback) && is_subclass_of($callback, \Throwable::class)) {
            $callback = fn ($e) => new $callback($e->getMessage(), $e);
        }

        $this->modelNotFoundResponseCallback = $callback;

        return $this;
    }

    /**
     * Build a Throwable from a ModelNotFoundException.
     *
     * @param  ModelNotFoundException  $e
     * @return \Throwable
     */
    protected function mapModelNotFoundException($e)
    {
        return call_user_func($this->modelNotFoundResponseCallback, $e);
    }
```